### ISSUE TITLE
Clicking pseudo elements

### ISSUE THREAD

**Issue by jonleighton** [teampoltergeist/poltergeist#281]
See https://github.com/thoughtbot/capybara-webkit/issues/494. Need to see if this affect Poltergeist.

**Comment by doppelreim** [comment]
Running into this issue as well using poltergeist v1.6.0 with capybara v2.4.4

Elements with a pseudo ::before-element can not be clicked.
Error:
```
Capybara::Poltergeist::MouseEventFailed:
       Firing a click at co-ordinates [490.5, 112.5] failed. Poltergeist detected
another element with CSS selector 'html body [...] ul.app-links li' at this position.
It may be overlapping the element you are trying to interact with. If you don't
care about overlapping elements, try using node.trigger('click').
```

It works, when I remove the ::before element.
The ::before-element is there to prepend an icon to links.

HTML:
```
<ul  class="app-links">
  <li>
    <a href="/foobar">Foobar</a>
  </li>
</ul>
```

spec:
```
click_on "Foobar"
```

What other information would be helpful to track down this thing?
Should I construct a complete minimal test-case that fails?

**Comment by anitagraham** [comment]
+1 with ::before prepending icons.

**Comment by anitagraham** [comment]
I have a minimal test case if anyone is interested.

**Comment by twalpole** [comment]
@anitagraham If you still have the minimal test case I would be interested in a link to it, also do you know if this is still an issue with phantomjs 2+ ?

**Comment by Bastes** [comment]
I have the same issue ; I fixed it following the "trigger(:click)" fix, but it would be nice for capybara to be able to work with links styled with ```:before``` pseudo-classes.

**Comment by twalpole** [comment]
@Bastes Do you have a minimal example that reproduces the issue?

**Comment by Bastes** [comment]
Hey  @twalpole ^^ here goes:

```css
a:before {
  content: "+"; /* Or whatever string you like */
}
```

```html
<a href="#" id="the-link">The Link</a>
```

```ruby
...
# these will fail:
click_on "The Link"
click_on "the-link"

# this will succeed:
find("a", text: "The Link").trigger("click")
...
```

Hope it helps ^^

**Comment by anitagraham** [comment]
I haven't tested it recently. I will try and find my test case!

**Comment by twalpole** [comment]
@Bastes ok -- so this fails on phantomjs 1.9.8 but works on 2.0

**Comment by twalpole** [comment]
This appears to work completely with PhantomJS 2, however with PhantomJS 1.9.8 there is an issue. PR#708 attempts to partly fix the issue with PhantomJS 1.9.8 in that it will now allow for clicking on an element that has a :before pseudo element as long the actual element has some size -- If the only piece of the element visible is the pseudo element then it still won't click it.  If anyone has any other ideas on how to completely fix this under 1.9.8 please take a look at PR #708 and give feedback.

**Comment by Bastes** [comment]
Thanks @twalpole ^^ cool news, in my case the element has a size indeed, but I'm on PhantomJS 1.9.0 (ubuntu 15.10...) so no surprise.

**Comment by twalpole** [comment]
This works correctly in phantomjs 2.1.1 --- Since there was no way to fully fix it in phantomjs < 2.0, the recommendation is to update.  Closing.
